### PR TITLE
#167727458  Update profile

### PR DESCRIPTION
**Pull request by O-Osahon**
### What does this PR do?
- It created the endpoint to enable users update their profiles

### Description of Task to be completed?
- Add failing tests for the `/api/v1/profile` endpoint
- Validate incoming data
- Write controller logic for the endpoint
- Create a factory function that returns any combination of data field(s) a user wants to update
- Write controller logic for the endpoint
- Add route and ensure the tests pass
- Document endpoint using Swagger

##### Using Postman:
-  If you haven't already, download [Postman](https://www.getpostman.com/)
- Download or clone this repository
- Run `git branch -r` to get all the branches from github
- Run `git checkout -b ft-profile-167727458 origin/ft-profile-167727458` to have a copy of this branch.
- Check the `.env.sample` file to know how to set your environment variables
- Run `npm run start:dev` The application should be running on `port 3000`
- Ensure you sign in via `localhost:3000/api/v1/auth/signup` using one of the `email` and `password` combination of a user's data in the `seed` folder. Or you could signup using your own data. 
- Set the `Authorization` header to the token that was returned during authentication.
- Checkout the API documentation at `localhost:3000/api-docs` to find out how to structure your requests
- Send a patch request to `localhost:3000/api/v1/profile`

##### Testing locally.
- After cloning and checking out this branch (as listed in the steps above), run `npm test`

### What are the relevant Pivotal Tacker stories?
[#167727458 ](https://www.pivotaltracker.com/story/show/167727458 )

### CODE REVIEW

**Review by G-Chilie — APPROVED** [review]

File: .env.example [hunk 1]
```diff
@@ -68,3 +68,5 @@ DEV_DB_URL
 DEV_DB_USER
 DEV_DB_PASSWORD
 DEV_DATABASE
+
+JWT_SECRET
```
Comment (okoroemeka) [inline .env.example]: a newline is required here.

Comment (O-Osahon) [reply .env.example]: resolved

**Comment by parkerthegeniuschild** [comment]
[x] Fix coverage drop
---

**Review by parkerthegeniuschild — DISMISSED** [review]
[x] Fix coverage drop
---

**Pull request merged**